src/pulp/solvers.py

```diff
@@ -2887,3 +2887,242 @@ def __init__(self, path=None, *args, **kwargs):
+            self.numvars = len(self.vars)
+            self.var_dict = {}
+            #Checking for repeated names
+            repeated_names = {}
+            for v in self.vars:
+                repeated_names[v.name] = repeated_names.get(v.name, 0) + 1
+            repeated_names = [(key, value) for key, value in list(repeated_names.items()) if value >= 2]
+            if repeated_names:
+                raise PulpError('Variables must have unique names for MOSEK-PuLP interface.')
+            #Creating a MOSEK environment
+            self.env = mosek.Env()
+            self.task = self.env.Task()
+            self.task.appendcons(self.numcons)
+            self.task.appendvars(self.numvars)
+            if self.msg:
+                self.task.set_Stream(mosek.streamtype.log,self.streampunk)
+            #Adding variables
+            for i in range(self.numvars):
+                vname = self.vars[i].name
+                self.var_dict[vname] = i
+                self.task.putvarname(i,vname)
+                #Variable type (Default: Continuous)
+                if self.mip & (self.vars[i].cat == LpInteger):
+                    self.task.putvartype(i,mosek.variabletype.type_int)
+                    self.solution_type = mosek.soltype.itg
+                #Variable bounds
+                vbkey = mosek.boundkey.fr
+                vup = inf
+                vlow = -inf
+                if self.vars[i].lowBound != None:
+                    vlow = self.vars[i].lowBound
+                    if self.vars[i].upBound != None:
+                        vup = self.vars[i].upBound
+                        vbkey = mosek.boundkey.ra
+                    else:
+                        vbkey = mosek.boundkey.lo
+                elif self.vars[i].upBound != None:
+                    vup = self.vars[i].upBound
+                    vbkey = mosek.boundkey.up 
+                self.task.putvarbound(i,vbkey,vlow,vup)
+                #Objective coefficient for the current variable.
+                self.task.putcj(i,lp.objective.get(self.vars[i],0.0))
+            #Coefficient matrix
+            self.A_rows,self.A_cols,self.A_vals = zip(*[[self.cons_dict[row],self.var_dict[col],coeff] 
+                                                        for col,row,coeff in lp.coefficients()])
+            self.task.putaijlist(self.A_rows,self.A_cols,self.A_vals)
+            #Constraints
+            self.constraint_data_list = []
+            row_i = 0
+            for c in self.cons:
+                cname = self.cons[c].name
+                if cname != None:
+                    self.task.putconname(self.cons_dict[c],cname)
+                else:
+                    self.task.putconname(self.cons_dict[c],c)
+                csense = self.cons[c].sense
+                cconst = -self.cons[c].constant
+                clow = -inf
+                cup = inf
+                #Constraint bounds
+                if csense == 0:
```

Comment:
here, it's better to use the names `LpConstraintLE`, `LpConstraintEQ` and `LpConstraintGE` for the constraints' senses